frontend/ui/src/app/tutorials/registry/registry-tutorial.component.html

```diff
@@ -0,0 +1,343 @@
+                      <p class="mt-1 text-xs">
+                        After clicking the button the new token will be displayed to you – This is the only time you
+                        will see it!
+                      </p>
+                    }
+                    <div class="mt-2 flex">
+                      @if (loginFormGroup.controls.tokenDone.value !== true) {
+                        <button
+                          type="button"
+                          [disabled]="loading()"
+                          (click)="createToken()"
+                          class="flex items-center justify-center rounded-lg border border-primary-700 bg-primary-700 px-5 py-2.5 text-sm font-medium text-white hover:border-primary-800 hover:bg-primary-800 focus:outline-none focus:ring-4 focus:ring-primary-300 dark:border-primary-700 dark:bg-primary-600 dark:hover:border-primary-700 dark:hover:bg-primary-700 dark:focus:ring-primary-800 sm:mt-0 sm:w-auto">
+                          Create New Token
+                        </button>
+                      }
+                    </div>
+
+                    @if (loginFormGroup.controls.tokenDone.value === true && createdToken; as t) {
+                      <div
+                        class="text-sm text-green-800 rounded-lg bg-green-50 dark:bg-gray-800 dark:text-green-400"
+                        role="alert">
+                        <p>
+                          Your Personal Access Token:
+                          <code class="select-all">{{ t.key }}</code>
+                          <app-clip class="mx-2" [clip]="t.key"></app-clip>
+                        </p>
+                        <p>
+                          <strong>Important:</strong>
+                          This is the only time you will be able to see this token, so please make sure to note it down
+                          before closing this page.
+                        </p>
+                      </div>
+                    }
+                  </li>
+
+                  <li class="mb-10 ms-8">
+                    <span class="absolute flex items-center justify-center -start-2">
+                      <input
+                        type="checkbox"
+                        [formControl]="loginFormGroup.controls.loginDone"
+                        class="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded-sm focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-700 dark:focus:ring-offset-gray-700 focus:ring-2 dark:bg-gray-600 dark:border-gray-500" />
+                    </span>
+                    <h3 class="text-lg font-medium leading-tight">Login</h3>
+                    <p class="mt-1 text-xs">
+                      Use the previously created Personal Access Token as password to log in to the registry with this
+                      command:
+                      <code class="mt-2 select-all block text-sm text-gray-900 dark:text-gray-200 whitespace-pre-line">
+                        echo "{{
```

Comment:
Is `echo "xxx" | docker login` really any better than `docker login -p xxx`?
If the concern is about storing the token in the shell history, it will be there either way, no?

Reply:
The thing is… it's not more secure, docker just doesn't know that :smile: